File: output/ChatGPT/codenet/C++/Python/s579031873.py
N, X = map(int, input().split())

ans = 0
min_val = 1000000000
for i in range(N):
    tmp = int(input())
    X -= tmp
    ans += 1
    if tmp < min_val:
        min_val = tmp

ans += X // min_val

print(ans)

File: output/ChatGPT/codenet/C++/Java/s579031873.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class s579031873 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N = Integer.parseInt(st.nextToken());
        int X = Integer.parseInt(st.nextToken());

        int ans = 0;
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < N; i++) {
            int tmp = Integer.parseInt(br.readLine());
            X -= tmp;
            ans++;
            if (tmp < min) min = tmp;
        }

        ans += X / min;

        System.out.println(ans);
    }
}